### ISSUE TITLE
Convert skin images into spikes

### ISSUE THREAD

**Issue by tehreemnaqvi** [fangwei123456/spikingjelly#196]
Hello,

I am trying to convert the image into spikes, but the results are weird. Could you please help me with this? You can find the input and the output image below. 

Your help will be appreciated.

Looking forward to your response.
Thank you!
![image](https://user-images.githubusercontent.com/38427571/161081051-4ede81cb-54e2-4c36-8b78-5e8bf85ddc64.png)
![image](https://user-images.githubusercontent.com/38427571/161081190-67766536-c971-476f-b72b-a6978977a2af.png)

**Comment by fangwei123456** [comment]
Hi, do you convert pixel values into range [0, 1]?

**Comment by tehreemnaqvi** [comment]
Thank you for your response. Yes, I normalized the images.

**Comment by tehreemnaqvi** [comment]
This is my code:

data = ImageFolder(root=data_path,
                   transform=transforms.Compose([transforms.ToTensor(), transforms.Resize([64, 64])]))

train_data, valid_data, test_data = random_split(data, [int(len(data) * 0.8), int(len(data) * 0.1),
                                                        len(data) - (int(len(data) * 0.8) + int(len(data) * 0.1))])

train_loader = DataLoader(train_data, batch_size=batch_size, shuffle=True, pin_memory=True)



      


for i, (images, labels) in enumerate(train_loader):
    for img, lbl in zip(images, labels):
        # print(img.shape)
        # print(lbl)
        eventForImg = np.zeros([1, 4096])
        for step in range(time_window):  # simulation time steps
            x = (torch.abs(img[0, :, :]) / 2) > torch.rand(img[0, :, :].size())
            #print(np.shape(x.type(torch.FloatTensor)))

            x1 = x[:, :].cpu()
            #print(x1.shape)
            x2 = np.array(x1)
            plt.figure(1)
            plt.ion()
            # plt.subplots(2, 2)
            # plt.title('Original Image')

            plt.imshow(img[0, :, :])
            plt.savefig("original.png")
            plt.axis('off')
            # plt.subplots(1, 2)
            plt.title('Original Image')
            plt.imshow(x2, cmap="hot")
            plt.axis('off')
            plt.gcf().canvas.draw()
            # plt.show()
            plt.savefig("melonoma.png")

**Comment by fangwei123456** [comment]
Can you upload a origin image file in zip format?

**Comment by tehreemnaqvi** [comment]
Yes, sure!

**Comment by tehreemnaqvi** [comment]
Images are not uploaded here as a zip file. Could you please give me your email address so I can share it over there?

**Comment by tehreemnaqvi** [comment]
[ISIC_0000002.zip](https://github.com/fangwei123456/spikingjelly/files/8395049/ISIC_0000002.zip)

**Comment by tehreemnaqvi** [comment]
[ISIC_0000036_downsampled.zip](https://github.com/fangwei123456/spikingjelly/files/8395064/ISIC_0000036_downsampled.zip)
[ISIC_0024403.zip](https://github.com/fangwei123456/spikingjelly/files/8395065/ISIC_0024403.zip)
[ISIC_0026343.zip](https://github.com/fangwei123456/spikingjelly/files/8395067/ISIC_0026343.zip)

**Comment by tehreemnaqvi** [comment]
Now, I successfully uploaded the single images.

**Comment by fangwei123456** [comment]
```python
import torch
from spikingjelly.clock_driven.encoding import PoissonEncoder
from torchvision import transforms
from PIL import Image
enc = PoissonEncoder()
to_tensor = transforms.ToTensor()
to_image = transforms.ToPILImage()
root = 'C:/Users/fw/Desktop/'
x = to_tensor(Image.open(root + 'ISIC_0000002.jpg'))  # [C, H, W]
y = enc(x)

for i in range(3):
    to_image(y[i]).save(root + f'{i}.bmp')
```
[img_spike.zip](https://github.com/fangwei123456/spikingjelly/files/8395280/img_spike.zip)

![image](https://user-images.githubusercontent.com/38352109/161209466-0b8f06c0-ac15-4894-863d-5f713b6df220.png)
![image](https://user-images.githubusercontent.com/38352109/161209476-f676e56f-5d41-4b64-b86c-2f324493449f.png)
![image](https://user-images.githubusercontent.com/38352109/161209483-235f4a24-2d07-4155-b284-517dc604dfcd.png)
![image](https://user-images.githubusercontent.com/38352109/161209487-087667b1-61d2-434a-8e1e-14d756351c22.png)

I suggest that you use `transforms.ToPILImage()` before you use `plt.imshow`.

**Comment by tehreemnaqvi** [comment]
Ahh, got it, Thank you. Is that output correct?

**Comment by fangwei123456** [comment]
Yes, I think the output spikes are correct. These spikes are similar to the origin image.

**Comment by tehreemnaqvi** [comment]
Yes, similar to input images. I usually use your functions for visualization, which are really helpful. 
Thank you so much for your help! Appreciated :)
